profile/src/main/java/rdx/works/profile/data/model/pernetwork/Network.kt

```diff
@@ -41,6 +41,15 @@ import rdx.works.profile.data.model.factorsources.Slip10Curve
 import rdx.works.profile.derivation.model.KeyType
 import rdx.works.profile.derivation.model.NetworkId
 
+interface EntityCreation {
+    fun createAccountWithBabylon(
+        displayName: String,
+        mnemonicWithPassphrase: MnemonicWithPassphrase,
+        deviceFactorSource: DeviceFactorSource,
+        onLedgerSettings: Network.Account.OnLedgerSettings
+    ): Network.Account
+}
+
```

Comment:
the refactor starts from here. We define an interface that the `Network` must implement. 
This will replace the `initAccountWithBabylonDeviceFactorSource`

Reply:
It also looks nice for creating unit tests